src/eckit/log/ChannelBuffer.cc

```diff
@@ -69,9 +67,10 @@ void ChannelBuffer::reset() {
 
 bool ChannelBuffer::dumpBuffer() {
     if (target_) {
-        target_->write(pbase(), pptr());
+        // Explicitly check that `pptr()` is not larger than end of buffer. Racecondition can end up adding larger values.
```

Comment:
Can we be explicit that this is a race condition in the std library, rather than here.